src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -137,6 +137,105 @@ class Context:
     symbol_map: dict[str, IteratorExpr | ValueExpr | SymbolExpr]
 
 
+def _builtin_neighbors_indirect_addressing(
+    transformer: "PythonTaskletCodegen", node: itir.Expr, node_args: list[itir.Expr]
+) -> list[ValueExpr]:
+    sdfg: dace.SDFG = transformer.context.body
+    state: dace.SDFGState = transformer.context.state
+
+    offset_literal, data = node_args
+    assert isinstance(offset_literal, itir.OffsetLiteral)
+    iterator = transformer.visit(data)
+    offset_dim = offset_literal.value
+    assert isinstance(offset_dim, str)
+    table: NeighborTableOffsetProvider = transformer.offset_provider[offset_dim]
+    assert isinstance(table, NeighborTableOffsetProvider)
+
+    shifted_dim = table.origin_axis.value
+
+    result_name = unique_var_name()
+    sdfg.add_array(result_name, dtype=iterator.dtype, shape=(table.max_neighbors,), transient=True)
+    result_access = state.add_access(result_name)
+
+    table_name = connectivity_identifier(offset_dim)
+    table_array = sdfg.arrays[table_name]
+
+    me, mx = state.add_map(
+        f"{offset_dim}_neighbors_map",
+        ndrange={"neigh_idx": f"0:{table.max_neighbors}"},
+    )
+    shift_tasklet = state.add_tasklet(
+        "shift",
+        code="__result = __table[__idx, neigh_idx]",
+        inputs={"__table", "__idx"},
+        outputs={"__result"},
+    )
+    data_access_tasklet = state.add_tasklet(
+        "data_access",
+        code="__result = __field[__idx]",
+        inputs={"__field", "__idx"},
+        outputs={"__result"},
+    )
+    idx_name = unique_var_name()
+    sdfg.add_scalar(idx_name, dace.int64, transient=True)
+    state.add_memlet_path(
+        state.add_access(table_name),
+        me,
+        shift_tasklet,
+        memlet=dace.Memlet(data=table_name, subset=",".join(f"0:{s}" for s in table_array.shape)),
+        dst_conn="__table",
+    )
+    state.add_memlet_path(
+        iterator.indices[shifted_dim],
+        me,
+        shift_tasklet,
+        memlet=dace.Memlet(data=iterator.indices[shifted_dim].data, subset="0"),
+        dst_conn="__idx",
+    )
+    state.add_edge(
+        shift_tasklet,
+        "__result",
+        data_access_tasklet,
+        "__idx",
+        dace.Memlet(data=idx_name, subset="0"),
+    )
+    state.add_memlet_path(
```

Comment:
Based on the assumption that `cell`/`edge` is always the 1st dim in shape, this code change in subset definition works in icon4py tests:
```suggestion
    field_subset = [f"0:{sdfg.arrays[iterator.field.data].shape[0]}"]
    if len(iterator.dimensions) > 1:
        field_subset += [f"i_{d}" for d in iterator.dimensions[1:]]
    state.add_memlet_path(
        iterator.field,
        me,
        data_access_tasklet,
        memlet=dace.Memlet(
            data=iterator.field.data,
            subset =",".join(field_subset),
        ),
        dst_conn="__field",
    )
```

Reply:
I don't think it's a good idea to assume that cell/edge will always be the first dimension.

Reply:
Right. I have addressed this comment in my latest commit.